src/main/java/gregtech/api/items/toolitem/ItemGTToolbelt.java

```diff
@@ -0,0 +1,762 @@
+        slotGroupWidget.debugName("toolbelt_inventory");
+        for (int i = 0; i < handler.getSlots(); i++) {
+            int finalI = i;
+            slotGroupWidget.child(new ItemSlot()
+                    .slot(SyncHandlers.itemSlot(handler, i).slotGroup(group)
+                            .changeListener(
+                                    (newItem, onlyAmountChanged, client, init) -> handler.onContentsChanged(finalI)))
+                    .background(GTGuiTextures.SLOT, GTGuiTextures.TOOL_SLOT_OVERLAY)
+                    .pos(i % 9 * 18, i / 9 * 18)
+                    .debugName("slot_" + i));
+        }
+        panel.child(slotGroupWidget);
+
+        return panel.bindPlayerInventory();
+    }
+
+    public static boolean isToolbeltableOredict(String oredict) {
+        return VALID_OREDICTS.contains(oredict);
+    }
+
+    public void registerValidOredict(String oredict) {
+        VALID_OREDICTS.add(oredict);
+    }
+
+    @Override
+    public float getDestroySpeed(@NotNull ItemStack stack, @NotNull IBlockState state) {
+        ItemStack selected = getHandler(stack).getSelectedStack();
+        if (selected != null) {
+            return selected.getItem().getDestroySpeed(selected, state);
+        } else return definition$getDestroySpeed(stack, state);
+    }
+
+    @Override
+    public boolean hitEntity(@NotNull ItemStack stack, @NotNull EntityLivingBase target,
+                             @NotNull EntityLivingBase attacker) {
+        ItemStack selected = getHandler(stack).getSelectedStack();
+        if (selected != null) {
+            return selected.getItem().hitEntity(selected, target, attacker);
+        } else return definition$hitEntity(stack, target, attacker);
+    }
+
+    @Override
+    public boolean onBlockStartBreak(@NotNull ItemStack itemstack, @NotNull BlockPos pos,
+                                     @NotNull EntityPlayer player) {
+        ItemStack selected = getHandler(itemstack).getSelectedStack();
+        if (selected != null) {
+            return selected.getItem().onBlockStartBreak(selected, pos, player);
+        } else return definition$onBlockStartBreak(itemstack, pos, player);
+    }
+
+    @Override
+    public boolean onBlockDestroyed(@NotNull ItemStack stack, @NotNull World worldIn, @NotNull IBlockState state,
+                                    @NotNull BlockPos pos, @NotNull EntityLivingBase entityLiving) {
+        ItemStack selected = getHandler(stack).getSelectedStack();
+        if (selected != null) {
+            return selected.getItem().onBlockDestroyed(selected, worldIn, state, pos, entityLiving);
+        } else return definition$onBlockDestroyed(stack, worldIn, state, pos, entityLiving);
+    }
+
+    @Override
+    public boolean getIsRepairable(@NotNull ItemStack toRepair, @NotNull ItemStack repair) {
+        // I think this lets repairs go through to the selected tool, in combination with the setDamage passthroughs?
+        // Idk testing required.
```

Comment:
It appears entirely functional, except that the anvil XP cost increase applies to the toolbelt instead of the tool. I don't know if we care about that at all.